### DISCUSSION TITLE
License of this library, GPL or LGPL?

### DISCUSSION THREAD

**Discussion by SpieringsAE** [analogdevicesinc/libiio#949] [Q&A]
I'm pretty new to the open source software environment, and 100% new to the legal side of it. And as I have a lot of respect for open source software and its creators I wish to handle this correctly.

I'm currently making a an open source project that uses this library, however I do not wish to use a GPL license on my project as I fear most of the people that use it would be strongly turned away by this. Under an LGPL license I can do this by using a dynamically linked library, but as I understand under a GPL license I can't.
The repository indicates both licenses so I'm not sure what my options are. Should I listen to the LGPL license which benefits me, or to the GPL license which hinders me?

To give a more clear picture of what is happening: I'm making a Matlab/Simulink blockset that will compile to C code and then machine code for an embedded Linux system which has IIO interfaces. It is this blockset that I wish to open source, but whatever someone else builds with it I would like to be free from the GPL license.

**Comment by rgetz** [comment]
@SpieringsAE : trying to get legal advice on github - is likely the wrong place.

As for your questions of libiio, there are two groups of files in this repo:
- LibIIO Library, Licensed under the LGPL
- LibIIO Tests/Examples, (under ./tests/* and ./examples/*), Licensed under the GPL.
- iiod, the libiio Deamon, 

What this means is you can use the library, under the LGPL, as long as your downstream users can replace/update the library when they want (either dynamically linking it, or by providing them the *.a files so they can rebuild things by hand - either is OK).

-Robin
